src/paperless/auth.py

```diff
@@ -2,12 +2,16 @@
 from django.contrib import auth
 from django.contrib.auth.middleware import PersistentRemoteUserMiddleware
 from django.contrib.auth.models import User
+from django.http import HttpRequest
 from django.utils.deprecation import MiddlewareMixin
 from rest_framework import authentication
 
 
 class AutoLoginMiddleware(MiddlewareMixin):
-    def process_request(self, request):
+    def process_request(self, request: HttpRequest):
+        # Dont use auto-login with token request
+        if request.path.startswith("/api/token/") and request.method == "POST":
```

Comment:
I'm wondering if it should be disabled for any request to `/api/` that is not a `GET` request. What do you think?

Any request that tries to modify data requires some CSRF token and would run into 5090, I guess.

Reply:
Or, another thought, just disable auto login for the whole `/api/`, and state that in the documentation.

Reply:
I'm no expert but I don't think we should do that. Again, auto-login acts as if its session auth which intentionally uses csrf. The frontend works of course because it correctly passes csrf and the api works with token auth, which doesn't require that. The exception here is that of course you need to be able to get the token

Reply:
I'll check one more thing when I'm in front of my machine later

Reply:
Ok, one more thing is the token auth should be above session auth, so this plan would still be my thought, if hitting the API directly use token auth, except in the case of /api/token/ which of course shouldn't require anything

Reply:
Thank you for fixing this!

I wish you some nice and relaxing holidays! 🎄